web/src/views/LoginView.jsx

```diff
@@ -1,29 +1,28 @@
 import { Form, Input, Button, Checkbox, Col, Row, message } from "antd";
 import { UserOutlined, LockOutlined } from "@ant-design/icons";
-import { Link, useLocation } from "react-router-dom";
-import { login } from "../api/api";
-import { useEffect } from "react";
+import { Link, useHistory } from "react-router-dom";
+import { login, getCurrentUser } from "../api/api";
+import { useState } from "react";
 
 export default function LoginView() {
+  const history = useHistory();
+
   const onFinish = values => {
     console.log("Sent values of form: ", values);
     login(values)
-      .then(() => {
+      .then(r => {
         message.success("Ok!");
+        getCurrentUser(r.data.access_token).then(r => {
+          history.push("/homepage", { message: "Logged in as " + r.data['logged_in_as'].username });
+        }).catch(r => {
+          if (r.response) message.error(r.response.data)
+        })
       })
       .catch(r => {
         if (r.response) message.error(r.response.data);
       });
-    // const user = getUserInformation()
   };
 
-  const location = useLocation();
-
-  useEffect(() => {
-    if (location.state && location.state.message) {
-      message.info(location.state.message);
-    }
-  }, [location]);
```

Comment:
We can keep this lines I guess. Assume after a user signs up, we can show a message, "Successfully signed up" in login page. I have seen those duplicate lines around the project. Maybe we can create a generic class so that pages can carry information using location.state ?

Reply:
I think we can get rid of the logged in messages completely and, update the homepage to display the currently-logged-in-username instead of the login button, which would redirect to the User Dashboard when clicked. We also need to add a logout button. I believe it would be better to create another issue to resolve these problems.

Reply:
I think their use-cases are a little different. Those messages usually indicate the result of a previous action such as login successful or login failed. We also use this logic in signup messages such as signed up succesfully, or user already exists etc. Of course we will update the homepage to display the profile/dashboard views based on login state.